oauthproxy.go

```diff
@@ -1051,6 +1051,8 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 	if p.PassAuthorization {
 		if session.IDToken != "" {
 			req.Header["Authorization"] = []string{fmt.Sprintf("Bearer %s", session.IDToken)}
+		} else if session.AccessToken != "" {
+			req.Header["Authorization"] = []string{fmt.Sprintf("Bearer %s", session.AccessToken)}
```

Comment:
@JoelSpeed I'll defer to you on whether to merge this or not since your refactor in #826 completely deletes this codepath.

Otherwise, seems sensible to me -- if we decide to move forward with this I'd like to see some unit tests added.

Reply:
I think given this would have to be considered a breaking change (people could start getting headers where they didn't before) and that we have the header work being done in #826 which would remove this anyway, we should not be accepting this change for now.

@kvaps For our next release we should have some alpha configuration that allows you to configure headers in a flexible way so that you can do this without the need for this addition.

The only case we haven't covered is fallback, ie, if the session field is empty, fallback to something else. I don't think this is desirable in most cases, unless you have something in particular in mind that justifies a fallback if a part of the session is empty?